### DISCUSSION TITLE
Discussion about GUI

### DISCUSSION THREAD

**Discussion by SinaKarvandi** [HyperDbg/HyperDbg#190] [General]
This is a discussion for discussing things related to GUI.

**Comment by ddkwork** [comment]
hello, i still can not test success with pname
[screenshots.zip](https://github.com/HyperDbg/HyperDbg/files/9116601/screenshots.zip)

**Comment by SinaKarvandi** [comment]
It couldn't be extracted. Would you please send the screenshot again?
![image](https://user-images.githubusercontent.com/13383992/179169966-63270b84-2592-40a7-a338-46909a63e6fd.png)

**Reply by ddkwork** [reply]
> 
[screenshots.gif.gz](https://github.com/HyperDbg/HyperDbg/files/9119616/screenshots.gif.gz)

**Comment by SinaKarvandi** [comment]
No, no, the script that I wrote was a simple static script to test.
You should use it in the command (event) script section.

For example: 
```
!cpuid script { 
	printf("process name: %s\n", $pname);
	
	printf("character 1 : %c (%x)\n", db($pname),  db($pname));
	printf("character 2 : %c (%x)\n", db($pname + 1), db($pname + 1));
	printf("character 3 : %c (%x)\n", db($pname + 2), db($pname + 2));
	printf("character 4 : %c (%x)\n", db($pname + 3), db($pname + 3));
	printf("character 5 : %c (%x)\n", db($pname + 4), db($pname + 4));
	
	if (db($pname) == 68 && db($pname + 1) == 79 && db($pname + 2) == 70 && db($pname + 3) == 65 && db($pname + 4) == 72) {
	
		printf("\nProcess name starts with: %c%c%c%c%c\n", 
				db($pname),
				db($pname + 1), 
				db($pname + 2),
				db($pname + 3),
				db($pname + 4));
	}
}
```

**Reply by ddkwork** [reply]
> No, no, the script that I wrote was a simple static script to test. You should use it in the command (event) script section.
> 
> For example:
> 
> ```
> !cpuid script { 
> 	printf("process name: %s\n", $pname);
> 	
> 	printf("character 1 : %c (%x)\n", db($pname),  db($pname));
> 	printf("character 2 : %c (%x)\n", db($pname + 1), db($pname + 1));
> 	printf("character 3 : %c (%x)\n", db($pname + 2), db($pname + 2));
> 	printf("character 4 : %c (%x)\n", db($pname + 3), db($pname + 3));
> 	printf("character 5 : %c (%x)\n", db($pname + 4), db($pname + 4));
> 	
> 	if (db($pname) == 68 && db($pname + 1) == 79 && db($pname + 2) == 70 && db($pname + 3) == 65 && db($pname + 4) == 72) {
> 	
> 		printf("\nProcess name starts with: %c%c%c%c%c\n", 
> 				db($pname),
> 				db($pname + 1), 
> 				db($pname + 2),
> 				db($pname + 3),
> 				db($pname + 4));
> 	}
> }
> ```

oh my ladygaga,  the pname ccan not include ext, it work now, thx.

**Comment by SinaKarvandi** [comment]
Hi, 
I hope you're well.

One thing I noticed from the GUI code is that you used the master branch as the project's base for loading drivers, modules, etc.

However, I made a clean and structured SDK for the HyperDbg (available at the Dev branch), which is completely finished now and can be used in the GUI.

Would you please test the GUI by using this SDK available at : 
https://github.com/HyperDbg/HyperDbg/tree/dev/hyperdbg/include/SDK

And also, an example of using it is available here:
https://github.com/HyperDbg/HyperDbg/blob/dev/hyperdbg/hyperdbg-cli/hyperdbg-cli.cpp

**Comment by SinaKarvandi** [comment]
Please also let us know whenever you need help or if there is anything that we should implement or change in order to ease the development of the GUI.

Currently, I don't know how to build the GUI project. It would be best if you could guide us on how to build the project, and after that, I can join to help you develop GUI. Let me know if there is anything I can do.

**Comment by ddkwork** [comment]
> Please also let us know whenever you need help or if there is anything that we should implement or change in order to ease the development of the GUI.
> 
> Currently, I don't know how to build the GUI project. It would be best if you could guide us on how to build the project, and after that, I can join to help you develop GUI. Let me know if there is anything I can do.

oh ha, sure. these days i am looking for the sdk about. i know it worked with deviceiocontrol api with in and out arg from some struct, so i am translate these codes now, please wait some days again. if the one event test sucess that the another events will be easily.

now worked these:
load driver  ---> passed with signed.

load vmm with event ---> doing

debug,request stack, mem ect event ---> doing.

so waiting please.
build the go language module is easily,
i will post a gif for you. wait me.

https://drive.google.com/file/d/1yXCa-gAh9LXfG98pZoYdltNoOhVAIiBy/view?usp=sharing

**Comment by SinaKarvandi** [comment]
That's okay. Thank you for all of your efforts on this project.

**Comment by SinaKarvandi** [comment]
> hello you,do you have wechat.?

No, I don't have a WeChat, but I can create an account there; send me your ID.

> go test sdk_test.go or click goland test button, use code gen code can be update right with your updates, i want`t change the c++ codes.

Honestly, I didn't get what you said. What do you mean by translating? Is it converting struct, defs, enums to GO? If yes then shouldn't we create a python script to convert them automatically?

Or let me say it another way. Are you wanna implement HPRDGCTRL.DLL, which is the main HyperDbg interface into the GO language. Isn't it better to use HPRDGCTRL.DLL inside our project as a wrapper to GO? I'm sure GO supports loading C/C++ modules.

Isn't it a better approach? what do you think about it?

**Comment by SinaKarvandi** [comment]
I mean, if you want we can easily export every functionality that you want, like reading, writing in registers, memory, whatever, into the HPRDGCTRL.DLL and you simply call it within GO. What do you think?

**Comment by SinaKarvandi** [comment]
Hi, 
I tried to create a WeChat account. However, this messenger seems to have weird conditions that won't let me sign up for it!
![image](https://user-images.githubusercontent.com/13383992/180600798-86956e61-973d-4228-9072-d31e1eafa1d7.png)

**Comment by SinaKarvandi** [comment]
However, it's pretty okay. I'll create a discussion in the GUI repo and we continue working on it.

We can continue our development discussion here : https://github.com/HyperDbg/gui/discussions/11

**Comment by ddkwork** [comment]
> However, it's pretty okay. I'll create a discussion in the GUI repo and we continue working on it.

try agin this 

https://tim.qq.com/mobile/index.html?adtag=index

discussion is slowly, i don't know when you online.

**Reply by SinaKarvandi** [reply]
I installed it. However, I don't understand anything from it. It's completely a Chinese application without any English description and it's probably not supported anywhere other than China.
![image](https://user-images.githubusercontent.com/13383992/180641408-842b4265-5de7-43c8-afb6-09b1f3ec5692.png)

**Reply by lingdianvision** [reply]
> > 但是，这没关系。我将在 GUI 存储库中创建一个讨论，我们将继续处理它。
> 
> 试试这个
> 
> https://tim.qq.com/mobile/index.html?adtag=index
> 
> 讨论很慢，我不知道你什么时候上网。

1437974052
